core/statedb/state_cache.go

```diff
@@ -110,6 +114,13 @@ func (c *StateCache) SetPendingUpdateAccount(accountIndex int64, account *types.
 	c.PendingUpdateAccountMap[accountIndex] = account
 }
 
+func (c *StateCache) SetPendingUpdateGasAccount(account *types.AccountInfo, assetId int64, balanceDelta *big.Int) {
+	if c.PendingGasAccount == nil {
+		c.PendingGasAccount = account
```

Comment:
`StateCache.PendingGasAccount` is a pointer which is same as `StateCache. PendingUpdateAccountMap[gasAccountIndex]`. If we update `StateCache. PendingUpdateAccountMap[gasAccountIndex]`, it will also influence `StateCache.PendingGasAccount` which is unexpected. We should only record gas asset delta for `GasAccount`

Reply:
fixed